internal/expr/bindtypes.go

```diff
@@ -174,50 +138,42 @@ func bindOutputTypes(ti typeNameToInfo, e *outputExpr) (outCols []columnAccessor
 		}
 
 		for _, t := range e.targetTypes {
-			info, ok := ti[t.typeName]
-			if !ok {
-				return nil, nil, typeMissingError(t.typeName, getKeys(ti))
-			}
 			if t.memberName == "*" {
 				// Generate asterisk columns.
-				allMembers, err := info.GetAllMembers()
+				members, memberNames, err := argInfo.AllOutputMembers(t.typeName)
```

Comment:
The `AllOutputMembers` method is on the `ArgInfo` struct, not actually on any interface. I think it does need a name change to something like `AllStructOutputs` because it does only work for structs, but the caller does not actually know that they are getting a struct back when they pass it the string with the name. In my opinion the method makes sense where it is but could do with a name change.

The error should also be changed, I didn't change this because  I didn't want to change the errors much in this PR but this one doesn't really make sense anymore.

For the code above, I don't think we should go back to exposing the information for individual info types to the `expr` package, but I think it should be explicit that a struct is expected.